java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1908,6 +1911,108 @@ public void zrangestore_by_lex(BaseClient client) {
+        assertArrayEquals(
+                new String[] {"one", "three", "two"}, client.zunion(new String[] {key1, key2}).get());
+        assertEquals(
+                Map.of("one", 1.0, "three", 3.0, "two", 5.5),
+                client.zunionWithScores(new String[] {key1, key2}).get());
+
+        // Union results are aggregated by the max score of elements
+        WeightAggregateOptions options =
+                WeightAggregateOptions.builder().aggregate(Aggregate.MAX).build();
+        assertArrayEquals(
+                new String[] {"one", "three", "two"},
+                client.zunion(new String[] {key1, key2}, options).get());
+        assertEquals(
+                Map.of("one", 1.0, "three", 3.0, "two", 3.5),
+                client.zunionWithScores(new String[] {key1, key2}, options).get());
+
+        // Union results are aggregated by the min score of elements
+        options = WeightAggregateOptions.builder().aggregate(Aggregate.MIN).build();
+        assertArrayEquals(
+                new String[] {"one", "two", "three"},
+                client.zunion(new String[] {key1, key2}, options).get());
+        assertEquals(
+                Map.of("one", 1.0, "two", 2.0, "three", 3.0),
+                client.zunionWithScores(new String[] {key1, key2}, options).get());
+
+        // Union results are aggregated by the sum of the scores of elements
+        options = WeightAggregateOptions.builder().aggregate(Aggregate.SUM).build();
+        assertArrayEquals(
+                new String[] {"one", "three", "two"},
+                client.zunion(new String[] {key1, key2}, options).get());
+        assertEquals(
+                Map.of("one", 1.0, "three", 3.0, "two", 5.5),
+                client.zunionWithScores(new String[] {key1, key2}, options).get());
+
+        // Scores are multiplied by 2.0 for key1 and key2 during aggregation.
+        options = WeightAggregateOptions.builder().weights(List.of(2.0, 2.0)).build();
+        assertArrayEquals(
+                new String[] {"one", "three", "two"},
+                client.zunion(new String[] {key1, key2}, options).get());
+        assertEquals(
+                Map.of("one", 2.0, "three", 6.0, "two", 11.0),
+                client.zunionWithScores(new String[] {key1, key2}, options).get());
+
+        // Union results are aggregated by the minimum score, with scores for key1 multiplied by 1.0 and
+        // for key2 by -2.0.
+        options =
+                WeightAggregateOptions.builder()
+                        .aggregate(Aggregate.MIN)
+                        .weights(List.of(1.0, -2.0))
+                        .build();
+        assertArrayEquals(
+                new String[] {"two", "three", "one"},
+                client.zunion(new String[] {key1, key2}, options).get());
+        assertEquals(
+                Map.of("two", -7.0, "three", -6.0, "one", 1.0),
+                client.zunionWithScores(new String[] {key1, key2}, options).get());
+
```

Comment:
The review checklist mentions tests should include a test for when keys don't exist, should we add that?

Reply:
They are treated as empty sets. But I can add it.